### PR TITLE
🐛 Fix duplicate deployment names in klusterlet-agent availability message

### PR DESCRIPTION
**Pull request by RokibulHasan7**
## Related issue(s)

Fixes #

### CODE REVIEW

File: pkg/operator/operators/klusterlet/controllers/statuscontroller/klusterlet_status_controller.go [hunk 1]
```diff
@@ -78,24 +78,27 @@ func (k *klusterletStatusController) sync(ctx context.Context, controllerContext
 	registrationDeploymentName := fmt.Sprintf("%s-registration-agent", klusterlet.Name)
 	workDeploymentName := fmt.Sprintf("%s-work-agent", klusterlet.Name)
 
-	if helpers.IsSingleton(klusterlet.Spec.DeployOption.Mode) {
+	IsSingleton := helpers.IsSingleton(klusterlet.Spec.DeployOption.Mode)
```
Comment (zhujian7) [inline pkg/operator/operators/klusterlet/controllers/statuscontroller/klusterlet_status_controller.go:81]:
```suggestion
	isSingleton := helpers.IsSingleton(klusterlet.Spec.DeployOption.Mode)
```

**Comment by zhujian7** [comment]
Another option is to deduplicate the agents in the `checkAgentsDeploymentAvailable` func, like:
```
// Check agent deployments, if both of them have at least 1 available replicas, return available condition
func checkAgentsDeploymentAvailable(ctx context.Context, kubeClient kubernetes.Interface, agents []klusterletAgent) metav1.Condition {
	var availableMessages []string
	components := sets.Set[string]{}  ### import "k8s.io/apimachinery/pkg/util/sets"
	for _, agent := range agents {
		componentID := fmt.Sprintf("%s-%s", agent.namespace, agent.deploymentName)
		if components.Has(componentID) {
			continue
		}
		components.Insert(componentID)
		deployment, err := kubeClient.AppsV1().Deployments(agent.namespace).Get(ctx, agent.deploymentName, metav1.GetOptions{})
		...
}

```

File: pkg/operator/operators/klusterlet/controllers/statuscontroller/klusterlet_status_controller.go (hunk 1 above)
Comment (RokibulHasan7) [reply pkg/operator/operators/klusterlet/controllers/statuscontroller/klusterlet_status_controller.go]: updated

**Comment by xuezhaojun** [comment]
@RokibulHasan7 Can we have a comparison of how the message looks before and after the change. This will help followers and code readers understand the fix better. Thank you!

**Comment by RokibulHasan7** [comment]
> @RokibulHasan7 Can we have a comparison of how the message looks before and after the change. This will help followers and code readers understand the fix better. Thank you!

Before:
```
  Type:			Available
  Status:		True
  LastTransitionTime:	2024-10-27 13:25:38 +0600 +06
  Reason:		KlusterletAvailable
  Message:		deployments are ready: klusterlet-agent,klusterlet-agent
  ```
  
After:
```
  Type:			Available
  Status:		True
  LastTransitionTime:	2024-10-27 13:25:38 +0600 +06
  Reason:		KlusterletAvailable
  Message:		deployments are ready: klusterlet-agent
```

**Comment by qiujian16** [comment]
good catch, thanks!
/approve

**Pull request merged**